ios/MaasNearbyTransit/Sources/MaasNearbyTransit/Components/NearbyTransitScheduleBadgesScrollView.swift

```diff
@@ -0,0 +1,214 @@
+import SwiftUI
+import Swappable
+
+private struct GradientScrollDataPreferenceKey: PreferenceKey {
+    typealias Value = ScrollViewData
+
+    static var defaultValue: ScrollViewData = .init(size: .zero, contentOffset: 0, contentSize: .zero)
+
+    static func reduce(value: inout ScrollViewData, nextValue: () -> ScrollViewData) {
+        value = nextValue()
+    }
+}
+
+private struct GradientBoundsScrollView<Content>: View where Content: View {
+
+    let content: Content
+
+    @State var trackableScrollData = ScrollViewData(size: .zero, contentOffset: .zero, contentSize: .zero)
+
+    init(@ViewBuilder content: () -> Content) {
+        self.content = content()
+    }
+
+    var body: some View {
+        GeometryReader { outsideProxy in
+            ScrollView(.horizontal, showsIndicators: false) {
+                ZStack(alignment: .leading) {
+
+                    HStack {
+                        content
+                    }
+
+                    GeometryReader { insideProxy in
+                        Color.clear.preference(
+                            key: GradientScrollDataPreferenceKey.self,
+                            value: ScrollViewData(
+                                size: outsideProxy.size,
+                                contentOffset: calculateContentOffset(fromOutsideProxy: outsideProxy, insideProxy: insideProxy),
+                                contentSize: insideProxy.size
+                            )
+                        )
+                    }
+                }
+            }
+            .onPreferenceChange(GradientScrollDataPreferenceKey.self) { trackableScrollData = $0 }
+            .overlay(
+                HStack(spacing: 0) {
+                    let opacities = gradientOpacities()
+                    gradient(at: outsideProxy.size.width / 2, opacity: opacities.0, reversed: false)
+                    gradient(at: -(outsideProxy.size.width / 2), opacity: opacities.1, reversed: true)
+                }
+            )
+        }
+        .frame(height: trackableScrollData.contentSize.height)
+    }
+
+    func gradient(at position: CGFloat, opacity: Double, reversed: Bool) -> some View {
+
+        let maxDiameter: CGFloat = 20
+
+        var colors: [Color] {
+            let colors = [Color(UIColor.systemBackground.withAlphaComponent(0)), Color(.systemBackground)]
```

Comment:
these colors will need to match exactly the color of the content background,
maybe worth exposing parameter for that 🤔

Reply:
Moving this in to separated pull request.